packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -342,7 +342,104 @@ describe('plugin-devices', () => {
       });
     });
 
-    describe('#getWebSocketUrl()', () => {});
+    describe('#getWebSocketUrl()', () => {
+      let services;
+
+      beforeEach('destructure services', () => {
+        services = webex.internal.services;
+      });
+
+      describe('when wait is truthy', () => {
+        let wait;
+
+        beforeEach('set wait', () => {
+          wait = true;
+        });
+
+        describe('when the device is registered', () => {
+          beforeEach('register the device', () => devices.register());
+
+          it('should resolve the promise with the websocket url',
+            () => devices.getWebSocketUrl(wait)
+              .then((url) => {
+                assert.isDefined(url);
+                assert.isTrue(services.isServiceUrl(url));
+                assert.include(url, 'mercury');
+              }));
+        });
+
+        describe('when the device is not registered', () => {
+          describe('when the device successfully registers', () => {
+            beforeEach(() => {
+              // This needs to not be returned, as it must be pending during the
+              // it clause.
+              devices.register();
```

Comment:
This seems like it could cause flakiness depending on speed of the test runner. I'd like to see this in the `it` block as maybe part of a `Promise.all` instead.

Reply:
I can see how this could cause flakiness. I have moved all unmanaged promises to their appropriate sections.